Dockerfile

```diff
@@ -1,9 +1,32 @@
+# Multistage dockerfile
+
+####
+# Stage 0:
+#
+# We need a requirements.txt, but we don't need pipenv installed in our final image,
+# so we do a multistage build where we generate the requirements.txt in our first stage,
+# and consume it in our final stage.
 FROM python:3.6
-WORKDIR /usr/src/app
-RUN pip install pipenv
+# Install pipenv.
+RUN python -m pip install pipenv
+# Copy Pipfile and Pipfile.lock.
 COPY Pipfile* /tmp/
+# Generate a requirements.txt file, which contains a list of all our project dependancies.
 RUN cd /tmp && pipenv lock --requirements > requirements.txt
-RUN pip install -r /tmp/requirements.txt
+
+####
+# Stage 1:
+#
+FROM python:3.6
+
+# Copy the requirements.txt from the 1st stage of the build.
+COPY --from=0 /tmp/requirements.txt /tmp
+# Install all our package requirements.
+RUN python -m pip install -r /tmp/requirements.txt
+
+WORKDIR /app
 COPY . .
+
 EXPOSE 8080
-CMD uvicorn main:app --reload --host 0.0.0.0 --port 8080
\ No newline at end of file
+
+CMD gunicorn main:APP -w 4 -k uvicorn.workers.UvicornWorker -b :8080
```

Comment:
I feel we need another Dockerfile where it reloads when the file system changes.

Reply:
You can override the cmd when you're doing development. Look at the Makefile,
`docker-compose run api uvicorn main:APP --reload --host 0.0.0.0 --port 8080`
So I don't think there's a need to create an entire new dockerfile. `docker-compose up` will give you a production style, but you can override that.

Reply:
Gotcha!